### ISSUE TITLE
Tokenizer allows inconsistent usage of tabs and spaces on the same indentation level

### ISSUE THREAD

**Issue by asleep-cult** [microsoft/pyright#2609]
**Describe the bug**
The tokenizer only validates the consistent use of tabs and spaces when they're on different levels of indentation. This results in a false positive when mixing tab indents with 8 space indents in the same block. This also means that mixing tab indents with 4 space indents (which looks the same in most IDEs) will result in an error message saying `Unindent amount does not match previous indent` (or 4 other errors depending on whether the tab or space indent comes first).

**To Reproduce**
1. Create a file
2. Write a block using 1 tab to indent the first line and 8 spaces to indent the second line
3. Run pyright on it

**Expected behavior**
Pyright should report an inconsistent use of tabs and spaces error.

**Screenshots or Code**
<img width="110" alt="Screenshot 2021-11-24 170818" src="https://user-images.githubusercontent.com/65962434/143318885-c1550a73-9da5-4f7d-a1a3-3164753f02c4.png">

**VS Code extension or command-line**
Command line pyright 1.1.188

**Comment by erictraut** [comment]
Thanks for reporting. The tokenizer did already catch some of these cases, but you've pointed out a case where it was not caught. I've addressed this hole. The fix will be included in the next release.

**Comment by asleep-cult** [comment]
> Thanks for reporting. The tokenizer did already catch some of these cases, but you've pointed out a case where it was not caught. I've addressed this hole. The fix will be included in the next release.

Does this change account for ambiguity in unindents as well?
<img width="140" alt="Screenshot 2021-11-24 192354" src="https://user-images.githubusercontent.com/65962434/143328685-34c56f49-d334-43e1-ab9f-0daa3f707a1b.png">
(GitHub renders this differently)
```py
if True:
	if False:  # 1 tab
		print("False")  # 2 tabs
        print("True")  # 8 spaces
```

**Comment by erictraut** [comment]
Ah, good catch. No, my previous change handled only the indent case. I've added additional logic to handle the dedent case as well.

**Comment by erictraut** [comment]
This is included in pyright 1.1.190, which I just published. It will also be included in the next release of pylance.
